Adyen/Security/SaleToPoiMessageSecuredEncryptor.cs

```diff
@@ -84,7 +86,42 @@ public string Decrypt(SaleToPoiMessageSecured saleToPoiMessageSecured, Encryptio
             var decryptedSaleToPoiMessageByteArray = _aesEncryptor.Decrypt(encryptedSaleToPoiMessageByteArray,
                                                                            encryptionDerivedKey,
                                                                            saleToPoiMessageSecured.SecurityTrailer.Nonce);
+            var receivedHmac = saleToPoiMessageSecured.SecurityTrailer.Hmac;
+            ValidateHmac(receivedHmac, decryptedSaleToPoiMessageByteArray, encryptionDerivedKey);
             return System.Text.Encoding.UTF8.GetString(decryptedSaleToPoiMessageByteArray);
         }
+
+        /// <summary>
+        /// Validate the hmac from a received message
+        /// </summary>
+        /// <param name="receivedHmac"></param>
+        /// <param name="decryptedSaleToPoiMessageByteArray"></param>
+        /// <param name="encryptionDerivedKey"></param>
+        private void ValidateHmac(byte[] receivedHmac, byte[] decryptedSaleToPoiMessageByteArray, EncryptionDerivedKey encryptionDerivedKey)
```

Comment:
tests?

Reply:
It is covered by the decryption test https://github.com/Adyen/adyen-dotnet-api-library/blob/develop/Adyen.Test/TerminalApiPosSecurityTest.cs#L63